engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateBy.java

```diff
@@ -409,8 +421,6 @@ private void computeCachedColumnRowsets(final Runnable resumeAction) {
                                         }
                                         // at least one dirty bucket will need this source
                                         srcNeeded = true;
-                                        // this window must be computed
-                                        dirtyWindows[winIdx] = true;
```

Comment:
We might be able to short-circuit some of the checks in this loop if we checked `dirtyWindows[winIdx]` before looking at the buckets. Also, I found it very confusing even on re-read that the window bucket context is called `winCtx` and the check method is `isWIndowDirty` not `isWindowBucketDirty` or similar.